engine-wasm/build-wasm.sh

```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
+set -euo pipefail
 
-cd "$PWD/src/main/resources/go"
-tinygo build -target=wasi -o ../compiled/cdc.wasm cdc.go
-cd "$OLDPWD"
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+docker run --rm \
+    -v ${SCRIPT_DIR}/src/main/resources:/src \
+    -w /src tinygo/tinygo:0.34.0 bash \
+    -c "tinygo build -ldflags='-extldflags --import-memory' --no-debug -target=wasm-unknown -o /tmp/cdc.wasm go/cdc.go && cat /tmp/cdc.wasm" > \
```

Comment:
3 notes on this line that's the core of the PR:

1. `-target=wasm-unknown` so that we don't need `wasi` to run the module (unless there is a reason to use wasi I'm missing)
2. with `-ldflags='-extldflags --import-memory'` we are telling TinyGo to emit a module that will need to be provided with an external Memory, hence, we are in full control of the Memory lifecycle
3. `&& cat /tmp/cdc.wasm` using stdout to avoid issues with file permissions going in/out the container